src/formatDate.js

```diff
@@ -50,7 +50,33 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  let result = '';
+  const dateParts = date.split(fromFormat[fromFormat.length - 1]);
+  const newDate = {};
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    newDate[fromFormat[i]] = dateParts[i];
+  }
+
+  if (newDate.YYYY) {
+    newDate.YY = newDate.YYYY.slice(-2);
+  }
+
+  if (newDate.YY < 30) {
+    newDate.YYYY = `20${newDate.YY}`;
+  } else {
+    newDate.YYYY = `19${newDate.YY}`;
+  };
+
+  for (let i = 0; i < toFormat.length - 1; i++) {
```

Comment:
Consider using 'for of' loop with entries() method instead of 'for' loop to iterate over toFormat array.